File: agent-service/src/services/textChunker.ts
/**
 * Split text into overlapping chunks for embedding.
 *
 * @param text       Source text to chunk
 * @param chunkSize  Approximate character count per chunk (default 500)
 * @param overlap    Number of characters to overlap between consecutive chunks (default 50)
 * @returns Array of chunk strings (non-empty)
 */
export function chunkText(
  text: string,
  chunkSize = 500,
  overlap = 50,
): string[] {
  if (!text || text.trim().length === 0) return [];

  const chunks: string[] = [];
  let start = 0;

  while (start < text.length) {
    const end = Math.min(start + chunkSize, text.length);
    const chunk = text.slice(start, end).trim();
    if (chunk.length > 0) {
      chunks.push(chunk);
    }
    // Advance by (chunkSize - overlap) so next chunk overlaps
    start += chunkSize - overlap;
    if (start >= text.length) break;
  }

  return chunks;
}

File: agent-service-py/src/agent_service/services/text_chunker.py
from __future__ import annotations


def chunk_text(text: str, chunk_size: int = 500, overlap: int = 50) -> list[str]:
    """
    Split ``text`` into overlapping chunks for embedding.

    :param text: Source text to chunk.
    :param chunk_size: Approximate character count per chunk.
    :param overlap: Number of characters to overlap between consecutive chunks.
    :returns: List of non-empty chunk strings.
    """
    if not text or not text.strip():
        return []

    chunks: list[str] = []
    start = 0

    while start < len(text):
        end = min(start + chunk_size, len(text))
        chunk = text[start:end].strip()
        if chunk:
            chunks.append(chunk)
        start += chunk_size - overlap
        if start >= len(text):
            break

    return chunks
